### ISSUE TITLE
never_loop does not detect `continue`s in let-else

### ISSUE THREAD

**Issue by mcpower** [rust-lang/rust-clippy#10031]
### Summary

Clippy seems to think that:

```rust
loop! {
  let ... = ... else {
    ...
    continue;
  };
  ...
  break;
}
```

never loops, even though there is a continue in a let-else.

(I'm relatively new to Rust - apologies if this isn't an actual bug!)

### Lint Name

clippy::never_loop

### Reproducer

A small example:
```
fn first_some<T>(v: Vec<Option<T>>) -> Option<T> {
    let mut it = v.into_iter();
    loop {
        let Some(option) = it.next() else {
            break None;
        };
        let Some(value) = option else {
            continue;
        };
        break Some(value);
    }
}
```

Clippy output:

```
error: this loop never actually loops
  --> src/lib.rs:3:5
   |
3  | /     loop {
4  | |         let Some(option) = it.next() else {
5  | |             break None;
6  | |         };
...  |
10 | |         break Some(value);
11 | |     }
   | |_____^
   |
   = note: `#[deny(clippy::never_loop)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#never_loop
```

I expected no error.

([Original code which caused this error](https://github.com/mcpower/adventofcode/blob/1ef7910fa2ddbba437d9181c498e33c1924d2e79/rust/src/lib.rs))

### Version

```text
rustc 1.65.0 (897e37553 2022-11-02)
binary: rustc
commit-hash: 897e37553bba8b42751c67658967889d11ecd120
commit-date: 2022-11-02
host: x86_64-unknown-linux-gnu
release: 1.65.0
LLVM version: 15.0.0
```

**Comment by mcpower** [comment]
Ah, dupe of #9356.
